nilearn/regions/parcellations.py

```diff
@@ -0,0 +1,429 @@
+
+        BaseDecomposition.__init__(self, n_components=200,
+                                   random_state=random_state,
+                                   mask=mask, memory=memory,
+                                   smoothing_fwhm=smoothing_fwhm,
+                                   standardize=standardize, detrend=detrend,
+                                   low_pass=low_pass, high_pass=high_pass,
+                                   t_r=t_r, target_affine=target_affine,
+                                   target_shape=target_shape,
+                                   mask_strategy='epi', mask_args=None,
+                                   memory_level=memory_level,
+                                   n_jobs=n_jobs,
+                                   verbose=verbose)
+
+    def fit(self, imgs, y=None, confounds=None):
+        """ Computes the mask and reduces the dimensionality of images
+        using randomized_svd. Then, fit the parcellation method on this
+        reduced data.
+
+        Parameters
+        ----------
+        imgs : List of Niimg-like objects
+            See http://nilearn.github.io/manipulating_images/input_output.html.
+            Data from which parcellations will be returned.
+
+        confounds : CSV file path or 2D matrix
+            Confounds to clean them from signals.
+            This parameter is passed to nilearn.signal.clean. Please see the
+            related documentation for details
+
+        Returns
+        -------
+        labels_ : numpy.ndarray
+            Labels to each cluster in the brain.
+
+        connectivity_ : numpy.ndarray
+            voxel-to-voxel connectivity matrix computed from a mask.
+            Note that, this attribute is returned only for selected methods
+            such as 'ward', 'complete', 'average'.
+        """
+        valid_methods = self.VALID_METHODS
+        if self.method is None:
+            raise ValueError("Parcellation method is specified as None. "
+                             "Please select one of the method in "
+                             "{0}".format(valid_methods))
+        if self.method is not None and self.method not in valid_methods:
+            raise ValueError("The method you have selected is not implemented "
+                             "'{0}'. Valid methods are in {1}"
+                             .format(self.method, valid_methods))
+
+        BaseDecomposition.fit(self, imgs)
+
+        data = mask_and_reduce(self.masker_, imgs, confounds,
+                               reduction_ratio='auto',
+                               n_components=self.n_components,
+                               random_state=self.random_state,
+                               memory_level=self.memory_level,
+                               n_jobs=self.n_jobs, memory=self.memory)
+
+        if self.verbose:
+            print("[Parcellations] Learning the data")
+        self._fit_method(data)
+
+        return self
+
+    def _fit_method(self, data):
```

Comment:
Why did you put this in a separate method? It would be simpler to inline it into the "fit" method.

Reply:
I thought it will be for easy readability.